tests/test_printer.py

```diff
@@ -535,7 +546,7 @@ def test_custom_format_attr():
     ctx.register_op(AnyOp)
     ctx.register_attr(CustomFormatAttr)
 
-    parser = Parser(ctx, prog)
+    parser = XDSLParser(ctx, prog, '<unknown>')
```

Comment:
IMO, this API looks quite bad (the `unknown` part). Could you not solve this with some default argument or something?

Reply:
Actually, init has a default constructor so this argument is not needed, right?

Reply:
Correct! I removed it from all places I could find!